### PR TITLE
fix(bhDateInterval): ensure date change is called

### LINKED ISSUE #3628: bhDateInterval has problem with minimal date for dateTo
When we change the date from manually, the minimum date is still the current date.

### PR DESCRIPTION
**Pull request by jniles**
This commit fixes a date change bug in the bhDateInterval that prevented updates to the minimum limit of the date.

Fixes #3628.

### CODE REVIEW

File: client/src/js/components/bhDateInterval.js [hunk 1]
```diff
@@ -62,20 +62,32 @@ function bhDateInterval(moment, bhConstants, Fiscal, Session) {
...
+    angular.extend($ctrl.pickerToOptions, { minDate : $ctrl.dateFrom });
+    if ($ctrl.onChange) {
+      $ctrl.onChange();
+    }
+  };
+
   function search(selection) {
     $ctrl.selected = selection.translateKey;
     selection.fn();
+    $ctrl.onChangeDate();
```
Comment (mbayopanda) [inline client/src/js/components/bhDateInterval.js:90]: Good 👍

File: client/src/js/components/bhDateInterval.js [hunk 2]
```diff
@@ -62,20 +62,32 @@ function bhDateInterval(moment, bhConstants, Fiscal, Session) {
...
+        $ctrl.pickerFromOptions.minDate = new Date(response.start_date);
+      });
+  }
+
+  $ctrl.onChangeDate = () => {
+    angular.extend($ctrl.pickerToOptions, { minDate : $ctrl.dateFrom });
+    if ($ctrl.onChange) {
+      $ctrl.onChange();
+    }
+  };
```
Comment (mbayopanda) [inline client/src/js/components/bhDateInterval.js:85]: Nice 👍

File: client/src/js/components/bhDateInterval.js [hunk 3]
```diff
@@ -62,20 +62,32 @@ function bhDateInterval(moment, bhConstants, Fiscal, Session) {
...
     // if controller has requested limit-min-fiscal, fetch required information
     if (angular.isDefined($ctrl.limitMinFiscal)) {
-      Fiscal.getEnterpriseFiscalStartDate(Session.enterprise.id)
-        .then((response) => {
-          $ctrl.pickerFromOptions.minDate = new Date(response.start_date);
-        });
+      getMinimumFiscalYearDate();
     }
 
     $ctrl.pickerToOptions = { showWeeks : false, minDate : $ctrl.dateFrom };
```
Comment (mbayopanda) [inline client/src/js/components/bhDateInterval.js]: I think this line must be before the ```if (angular.isDefined($ctrl.limitMinFiscal)) {```

**Review by mbayopanda — APPROVED** [review]
The code LGTM and works fine

I just noticed that in the fiscal year module, we have to click twice on date key (today, this week, this month, this year) for seeing change for `Nombre de mois`

File: client/src/js/components/bhDateInterval.js (hunk 3 above)
Comment (jniles) [reply client/src/js/components/bhDateInterval.js]: Good point!

**Comment by jniles** [comment]
@mbayopanda thanks for the review.  I'll fix what you point out and see if I can fix the issues with the FY module.

**Comment by jniles** [comment]
Looks like this passes now.

bors r+

**Pull request merged**